src/pydap/simple_server.py

```diff
@@ -0,0 +1,92 @@
+from webob.request import Request
+import threading
+import time
+
+from werkzeug.serving import run_simple
+from .handlers.csv import CSVHandler
+from .wsgi.ssf import ServerSideFunctions
+
+
+def run_simple_server(test_file, port=8000, handler=CSVHandler):
+    application = handler(test_file)
+    application = ServerSideFunctions(application)
+
+    def app_check_for_shutdown(environ, start_response):
+        if environ['PATH_INFO'].endswith('shutdown'):
+            shutdown_server(environ)
+            return shutdown_application(environ, start_response)
+        else:
+            return application(environ, start_response)
+
+    run_simple('0.0.0.0', port,
+               app_check_for_shutdown)
+
+
+def shutdown_server(environ):
+    if 'werkzeug.server.shutdown' not in environ:
+        raise RuntimeError('Not running the development server')
+    environ['werkzeug.server.shutdown']()
+
+
+def shutdown_application(environ, start_response):
+    start_response('200 OK', [('Content-Type', 'text/plain')])
+    return ['Server is shutting down.']
+
+
+class LocalTestServer:
+    """
+    Simple server instance that can be used to test pydap.
+    Relies on threading and is usually slow (it has to
+    start and shutdown which typically takes ~2 sec).
+
+    Usage:
+    >>> from pydap.handlers.csv import CSVHandler
+    >>> from pydap.client import open_url
+    >>> server = LocalTestServer("test.csv", handler=CSVHandler)
+    >>> server.connect()
+    >>> dataset = open_url("http://localhost:%s" % server.port)
+    >>> server.disconnect()
+
+    Or as an instance:
+    >>> with LocalTestServer("test.csv", handler=CSVHandler) as server:
+    >>>     dataset = open_url("http://localhost:%s" % server.port)
+    """
+
+    def __init__(self, file_name, port=8000,
```

Comment:
Ok, I've figured it out. It seems to work and, yes, that would be much more robust!